ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixClusterManager.java

```diff
@@ -1408,6 +1474,27 @@ private List<ReplicaId> initializeDisksAndReplicasOnNode(AmbryDataNode datanode,
       return addedReplicas;
     }
 
+    /**
+     * Remove the disks and replicas on the given node. Remove associated partitions if this is the only partition.
+     * @param datanode the {@link AmbryDataNode} that is being initialized.
+     * @return a list of replicas present on this deleted node.
+     */
+    private List<ReplicaId> removeDisksAndReplicasOnNode(AmbryDataNode datanode) {
```

Comment:
this can also be folded in the caller. Write comments because the top caller will be big when u combine all these into it.

Reply:
Done